sqlmesh/core/model/definition.py

```diff
@@ -934,6 +935,7 @@ def metadata_hash(self) -> str:
                 self.project,
                 str(self.allow_partials),
                 gen(self.session_properties_) if self.session_properties_ else None,
+                str(self.optimize) if self.optimize is not None else None,
```

Comment:
Not sure why we need the branch here:
```suggestion
                str(self.optimize),
```

Reply:
Reverted that entirely